### PR TITLE
chore: Introduce FlatMetricFamily struct to flatten prometheus metric output

### PR DESCRIPTION
**Pull request by jeffreylimnardy**
## Description

Changes proposed in this pull request (what was done and why):

- Similar to #1371, Prometheus output logs are deeply nested and unreadable. With this PR, a FlatMetricFamily struct is introduced
- FlatMetricFamily makes output from self-monitor tests more readable and concise

Changes refer to particular issues, PRs or documents:

- 

## Traceability
- [ ] The PR is linked to a GitHub issue.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [ ] If the change is user-facing, the documentation has been adjusted.
- [ ] If a CRD is changed, the corresponding Busola ConfigMap has been adjusted.
- [ ] The feature is unit-tested.
- [ ] The feature is e2e-tested.

### CODE REVIEW

File: test/testkit/matchers/prometheus/prommetric_utils.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
+package prometheus
+
+import prommodel "github.com/prometheus/client_model/go"
+
+// FlatMetricFamily holds all necessary information about a prometheus MetricFamily.
+// Gomega doesn't handle deeply nested data structure very well and generates large, unreadable diffs when paired with
+// the deeply nested structure of pmetrics.
```
Comment (skhalash) [inline test/testkit/matchers/prometheus/prommetric_utils.go:7]: A bit of copy-paste here 😄 It has nothing to do with pmetrics

File: test/testkit/matchers/prometheus/prommetric_utils.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+//
+// Introducing a go struct with a flat data structure by extracting necessary information from different levels of
+// metricfamily makes accessing the information easier and improves readability of the output.
```
Comment (skhalash) [inline test/testkit/matchers/prometheus/prommetric_utils.go:10]:
```suggestion
// FlatMetricFamily is a flat data structure that provides necessary information from different levels of
// MetricFamily makes accessing the information easier and improves readability of the Gomega output.
```

File: test/testkit/matchers/prometheus/prommetric_utils.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+type FlatMetricFamily struct {
+	Name         string
+	MetricValues float64
+	Labels       map[string]string
+}
+
+// flattenAllMetricFamily flattens an array of prometheus MetricFamily to a slice of FlatMetricFamily.
+// It converts the deeply nested MetricFamily to a flat struct, making it more readable in the test output.
+func flattenAllMetricFamily(mfs map[string]*prommodel.MetricFamily) []FlatMetricFamily {
```
Comment (skhalash) [inline test/testkit/matchers/prometheus/prommetric_utils.go:19]:
```suggestion
func flattenAllMetricFamilies(mfs map[string]*prommodel.MetricFamily) []FlatMetricFamily {
```

File: test/testkit/matchers/prometheus/prommetric_utils.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+// FlatMetricFamily holds all necessary information about a prometheus MetricFamily.
+// Gomega doesn't handle deeply nested data structure very well and generates large, unreadable diffs when paired with
+// the deeply nested structure of pmetrics.
+//
+// Introducing a go struct with a flat data structure by extracting necessary information from different levels of
+// metricfamily makes accessing the information easier and improves readability of the output.
+type FlatMetricFamily struct {
+	Name         string
+	MetricValues float64
```
Comment (skhalash) [inline test/testkit/matchers/prometheus/prommetric_utils.go:13]: Is it multiple values or a single one?

File: test/testkit/matchers/prometheus/prommetric_utils.go [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+import prommodel "github.com/prometheus/client_model/go"
+
+// FlatMetricFamily holds all necessary information about a prometheus MetricFamily.
+// Gomega doesn't handle deeply nested data structure very well and generates large, unreadable diffs when paired with
+// the deeply nested structure of pmetrics.
+//
+// Introducing a go struct with a flat data structure by extracting necessary information from different levels of
+// metricfamily makes accessing the information easier and improves readability of the output.
+type FlatMetricFamily struct {
```
Comment (skhalash) [inline test/testkit/matchers/prometheus/prommetric_utils.go:11]: Would it be possible to also add the metric type?

**Review by skhalash — CHANGES_REQUESTED** [review]

File: test/testkit/matchers/prometheus/prommetric_utils.go (hunk 1 above)
Comment (jeffreylimnardy) [reply test/testkit/matchers/prometheus/prommetric_utils.go]: Missed that one 😅

**Review by skhalash — APPROVED** [review]

**Pull request merged**